### PR TITLE
Update repo structure operationalinsight

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Move dataplane libraries to new sdk directory.
Update directory Structure.

### CODE REVIEW

**Comment by chidozieononiwu** [comment]
@Azure/azure-sdk-eng

**Comment by chidozieononiwu** [comment]
cc @shahabhijeet

File: sdk/Directory.Build.props [hunk 1]
```diff
@@ -3,5 +3,10 @@
     <IsMgmtProject Condition="'$(IsMgmtProject)' == '' and $(MSBuildProjectName.Contains('.Management.'))">true</IsMgmtProject>
     <IsDataPlaneProject Condition="'$(IsDataPlanProject)' == '' and $(MSBuildProjectName.StartsWith('Microsoft.Azure.')) and '$(IsMgmtProject)' != 'true'">true</IsDataPlaneProject>
   </PropertyGroup>
+
+  <PropertyGroup>
```
Comment (weshaggard) [inline sdk/Directory.Build.props]: Why are these here? We should be converting these to the data-plane style projects. If we end up needing some of these properties for some reasons they should go into the eng/Directory.Build.Data.targets|props files.

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  <PropertyGroup>
+    <Description>Microsoft Azure Operational Insights Data Library</Description>
+    <PackageId>Microsoft.Azure.OperationalInsights</PackageId>
+    <Description>Provides Microsoft Azure Operational Insights data operations</Description>
+    <AssemblyName>Microsoft.Azure.OperationalInsights</AssemblyName>
+    <VersionPrefix>0.10.0-preview</VersionPrefix>
+    <PackageTags>Microsoft Azure OperationalInsights;OperationalInsights</PackageTags>    
+	  <PackageReleaseNotes>Taking dependency on 10.0.3 version of Newtonsoft nuget package.</PackageReleaseNotes>
+
+    <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
```
Comment (weshaggard) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj]: Similar to my comments in HDInsights PR we shouldn't be setting these but instead we should be fixing them to work with the data-plane targets.

**Comment by weshaggard** [comment]
@shahabhijeet do you know who the owner of this library is? We would like to make sure they are aware we are moving it.

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj (hunk 2 above)
Comment (chidozieononiwu) [reply sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj]: Ok, I am updating this now.

File: sdk/Directory.Build.props (hunk 1 above)
Comment (chidozieononiwu) [reply sdk/Directory.Build.props]: Will update this.

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>Microsoft Azure Operational Insights Data Library</Description>
```
Comment (weshaggard) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj]: You have description twice.

File: eng/AzSdk.test.reference.targets [hunk 4]
```diff
@@ -12,4 +12,6 @@
     <!-- This is needed for discovering tests in test explorer -->
     <PackageReference Include="System.Runtime.InteropServices" Version="4.3.0" />
   </ItemGroup>
+
```
Comment (weshaggard) [inline eng/AzSdk.test.reference.targets]: Please undo this change.

File: sdk/Directory.Build.props [hunk 5]
```diff
@@ -3,5 +3,6 @@
     <IsMgmtProject Condition="'$(IsMgmtProject)' == '' and $(MSBuildProjectName.Contains('.Management.'))">true</IsMgmtProject>
     <IsDataPlaneProject Condition="'$(IsDataPlanProject)' == '' and $(MSBuildProjectName.StartsWith('Microsoft.Azure.')) and '$(IsMgmtProject)' != 'true'">true</IsDataPlaneProject>
   </PropertyGroup>
+
```
Comment (weshaggard) [inline sdk/Directory.Build.props]: Please undo this change.

File: eng/AzSdk.reference.targets [hunk 6]
```diff
@@ -3,7 +3,7 @@
     <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure" Version="[3.3.18, 4.0.0)" />
     <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="[2.3.18, 3.0.0)" />
   </ItemGroup>
-
+
```
Comment (weshaggard) [inline eng/AzSdk.reference.targets]: Please undo this change.

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    <Description>Provides Microsoft Azure Operational Insights data operations</Description>
+    <AssemblyTitle>Microsoft Azure Operational Insights</AssemblyTitle>
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
```
Comment (weshaggard) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj]: As we chatted please add the version. Looking at the old project it seems as though we are missing the following:
```
    <VersionPrefix>0.10.0-preview</VersionPrefix>
    <PackageTags>Microsoft Azure OperationalInsights;OperationalInsights</PackageTags>    
	  <PackageReleaseNotes>Taking dependency on 10.0.3 version of Newtonsoft nuget package.</PackageReleaseNotes>
```

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  </PropertyGroup>
```
Comment (weshaggard) [inline sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj]: You changed the TargetFramework list from <SdkTargetFx>net452;net461;netstandard1.4;netstandard2.0</SdkTargetFx> to RequiredTargetFrameworks. We need to be sure we are including the correct set of target frameworks. Can you please file an issue tracking this? I suspect we will want to update RequiredTargetFrameworks to include these extra frameworks that SdkTargetFx includes.

File: eng/AzSdk.reference.targets [hunk 9]
```diff
@@ -3,7 +3,7 @@
     <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure" Version="[3.3.18, 4.0.0)" />
     <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="[2.3.18, 3.0.0)" />
   </ItemGroup>
-
+  
   <ItemGroup Condition="'$(IsTargetingNetStandard)' == 'true'">
```
Comment (weshaggard) [inline eng/AzSdk.reference.targets]: Looks like there is still some whitespaces changes in this file.

Comment (chidozieononiwu) [reply eng/AzSdk.reference.targets]: it's exactly the same as in the master branch though.

Comment (weshaggard) [reply eng/AzSdk.reference.targets]: I see an extra space in the diff here on github.

File: sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj (hunk 8 above)
Comment (chidozieononiwu) [reply sdk/operationalinsights/Microsoft.Azure.OperationalInsights/src/Microsoft.Azure.OperationalInsights.csproj]: I have filed an issue for that https://github.com/Azure/azure-sdk-for-net/issues/5959

File: eng/AzSdk.reference.targets (hunk 9 above)
Comment (chidozieononiwu) [reply eng/AzSdk.reference.targets]: Oh, so sorry. I have fixed that now.

**Review by weshaggard — APPROVED** [review]
